src/Validator/Parser/hdtXSDSchemaParser.cpp

```
@@ -0,0 +1,413 @@
+							attrCons[aName] = tc;
+					}
+				}
+				gatherComplexTypeAttrs(child, namedSimpleTypes, attrCons);
+			}
+		}
+
+		/// Parses all top-level elements and returns their allowed enumeration values.
+		std::unordered_map<std::string, std::unordered_set<std::string>>
+			parseAllElementEnumerations(const pugi::xml_document& doc)
+		{
+			pugi::xml_node schema = doc.first_child();
```

Clang (14.0.6)

[warning] variable 'schema' is not initialized

(cppcoreguidelines-init-variables)